### PR TITLE
Merged delayed withdrawal with same token to comply with protocol

### LINKED ISSUE #514: Merge Delayed Withdrawals from same token
Pending Withdrawals for Delayed Withdrawals for the  same token need to be merged. When you do the final withdraw from the DelayedWithdrawal smart contract, it withdraws the total balance for that token for that user. Also, the timer is reset everytime a nonInstant withdraw for the same token occurs

We'll need to filter out exits for the same token that have had a nonInstant withdraw done and merge them. We can either merge them using the info from pendingDelayedWithdraws in Storage or fetching the information from the DelayedWithdrawal smart contract which returns the information merged.

Fetching the information from the smart contract would also help reduce our dependency on the local storage to track the delayed withdraw..

### PR DESCRIPTION
**Pull request by AlbertoElias**
Closes #514 

### What does this PR does?

Merged delayed withdrawal with same token to comply with protocol. This is done in Home, Account Details and Transaction Overview. Also handles deleting txs from storage as needed.

### How to test?

Making multiple Exits that will trigger Delayed Withdrawals of the same token

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Respect code style and lint
- [x] Update documentation (if needed)

### CODE REVIEW

File: src/store/global/global.thunks.js [hunk 1]
```diff
@@ -291,8 +291,20 @@ function checkPendingDelayedWithdrawals () {
             revertedTxReceipts.forEach((tx) => {
               dispatch(removePendingDelayedWithdrawByHash(tx.transactionHash))
             })
+
+            // Checks with Coordinator API if exit has been withdrawn
+            const exitsApiPromises = accountPendingDelayedWithdraws.map((pendingDelayedWithdraw) => {
+              return CoordinatorAPI.getExit(pendingDelayedWithdraw.batchNum, pendingDelayedWithdraw.accountIndex)
+                .then((exitTx) => {
+                  // Checks here to have access to pendingDelayedWithdraw.hash
```
Comment (elias-garcia) [inline src/store/global/global.thunks.js:299]: I think that `hash` in this comment should be `id`

File: src/store/transaction/transaction.reducer.js [hunk 2]
```diff
@@ -234,7 +234,7 @@ function transactionReducer (state = initialTransactionState, action) {
             ...state.steps[STEP_NAME.REVIEW_TRANSACTION],
             transaction: {
               exit: action.exit,
-              amount: action.exit.balance,
+              amount: action.exit.balance || action.exit.amount,
```
Comment (elias-garcia) [inline src/store/transaction/transaction.reducer.js:237]: Why an exit can have `balance` and `amount`?

File: src/store/transaction/transaction.thunks.js [hunk 3]
```diff
@@ -98,6 +101,15 @@ function fetchExit (accountIndex, batchNum) {
...
       CoordinatorAPI.getExit(batchNum, accountIndex)
     ]).then(([account, exit]) => {
+      // If we are completing a delayed withdrawal, we need to merge all delayed withdrawals
+      // of the same token to show the correct amount in Transaction Overview
+      if (completeDelayedWithdrawal) {
+        const pendingDelayedWithdrawsWithToken = pendingDelayedWithdraws
+          .filter((pendingDelayedWithdraw) => pendingDelayedWithdraw.token.id === exit.token.id)
+        const mergedPendingDelayedWithdraws = mergeDelayedWithdraws(pendingDelayedWithdrawsWithToken)
+        exit = mergedPendingDelayedWithdraws[0]
+        console.log(pendingDelayedWithdrawsWithToken, mergedPendingDelayedWithdraws, exit)
```
Comment (elias-garcia) [inline src/store/transaction/transaction.thunks.js:111]: Leftover here

File: src/store/transaction/transaction.reducer.js (hunk 2 above)
Comment (AlbertoElias) [reply src/store/transaction/transaction.reducer.js:237]: Balance is when we use the Exit object returned by the API, amount when it comes from the storage 😔

**Review by elias-garcia — APPROVED** [review]

**Review by elias-garcia — APPROVED** [review]

**Pull request merged**